In [13]:
from math import sqrt

def normalized_v(i,j,k):
    if (i+j+k)%2 == 0:
        V = 1/sqrt(i**2+j**2+k**2)
    else:
        V = -1/sqrt(i**2+j**2+k**2)
    return V

max_steps = int(input('Enter the size of the integer L'))
s=0.

for i in range(-max_steps,max_steps+1):
    for j in range(-max_steps,max_steps+1):
        for k in range(-max_steps,max_steps+1):
            if (i==0 and j==0 and k==0):
                continue
            else:
                s += normalized_v(i,j,k)
            
print('The Madelung constant M is" ',s)

Enter the size of the integer L 600


The Madelung constant M is"  -1.7466031453918351


First Attempt $L=100$: $-1.7418198158396654$

Second Attempt $L=200$: $-1.7446850421707383$

Third Attempt $L=300$: $-1.7456432959126562$

Fourth Attempt $L=400$: $-1.746123021009531$

Fifth Attempt $L=600$: $-1.7466031453918351$







